### PR TITLE
New Major API

### PR DESCRIPTION
**Pull request by gamburgm**
Following the new spec, which is attached below:

```graphql
type MajorQuery extends Query {
  major(majorId: String!): Major
}

type Major {
  majorId: String!

  occurrence(year: Int!): MajorOccurrence!
  latestOccurrence: MajorOccurrence!
}

type MajorOccurrence {
  majorId: String!
  yearVersion: String!

  spec: JSON!
  plansOfStudy: JSON!
}
```

### CODE REVIEW

File: backend/graphql/tests/__snapshots__/major.test.seq.ts.snap [hunk 1]
```diff
@@ -2,11 +2,9 @@
 
 exports[`cannot find major from non-present majorId 1`] = `
 Object {
-  "data": Object {
-    "major": null,
-  },
+  "data": undefined,
   "errors": Array [
-    [GraphQLError: major not found!],
+    [ValidationError: Cannot query field "majorId" on type "Query". Did you mean "major"?],
```
Comment (megandouli) [inline backend/graphql/tests/__snapshots__/major.test.seq.ts.snap]: Is this ValidationError supposed to be in the snapshot?

File: backend/graphql/tests/__snapshots__/major.test.seq.ts.snap [hunk 2]
```diff
@@ -19,14 +17,9 @@ Object {
...
-  "data": Object {
-    "major": Object {
-      "name": "Computer Science",
-      "occurrence": null,
-    },
-  },
+  "data": undefined,
   "errors": Array [
-    [GraphQLError: occurrence not found!],
+    [ValidationError: Cannot query field "majorId" on type "Query". Did you mean "major"?],
```
Comment (megandouli) [inline backend/graphql/tests/__snapshots__/major.test.seq.ts.snap]: ^^ Same as above

File: scripts/migrate_major_data.ts [hunk 3]
```diff
@@ -1,40 +1,51 @@
...
+      }
     });
-  });
+  }, { concurrency: CONCURRENCY_COUNT });
 }
 
-migrateData(fetchData('major.json') as MajorJSON);
+(async () => {
+  await migrateData(fetchData().all_objects);
+  console.log('Success! You may close.');
```
Comment (megandouli) [inline scripts/migrate_major_data.ts]: Do we want to also log the amount of time it took to migrate all the data?

**Review by megandouli — APPROVED** [review]
Overall looks good! Just want to confirm the test snapshot is what we want

**Review by dajinchu — APPROVED** [review]

File: backend/graphql/tests/__snapshots__/major.test.seq.ts.snap (hunk 1 above)
Comment (gamburgm) [reply backend/graphql/tests/__snapshots__/major.test.seq.ts.snap]: WOW, great catch, just fixed this.

File: scripts/migrate_major_data.ts (hunk 3 above)
Comment (gamburgm) [reply scripts/migrate_major_data.ts]: Great idea!

**Review by megandouli — APPROVED** [review]
chungus

**Pull request merged**